src/main/java/chatbot/Parser.java

```diff
@@ -0,0 +1,184 @@
+
+        try {
+            nextInput = input.substring(4);
+
+            if (nextInput.isBlank()) {
+                throw new InvalidCommandException();
+            }
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new InvalidCommandException();
+        }
+
+        if (nextInput.startsWith("todo")) {
+            try {
+                String description = nextInput.substring(5);
+
+                if (description.isBlank()) {
+                    throw new InvalidTodoException();
+                } else {
+                    arr.add(0, "todo");
+                    arr.add(1, description);
+                }
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new InvalidTodoException();
+            }
+        } else if (nextInput.startsWith("deadline")) {
+            try {
+                int by = nextInput.indexOf("/by");
+                String description = nextInput.substring(9, by - 1);
+                String deadline = nextInput.substring(by + 4);
+
+                if (description.isBlank() || deadline.isBlank()) {
+                    throw new InvalidDeadlineException();
+                } else {
+                    arr.add(0,"deadline");
+                    arr.add(1, description);
+                    arr.add(2, deadline);
+                }
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new InvalidDeadlineException();
+            }
+        } else if (nextInput.startsWith("event")) {
+            try {
+                int from = nextInput.indexOf("/from");
+                int to = nextInput.indexOf("/to");
+                String description = nextInput.substring(6, from - 1);
+                String start = nextInput.substring(from + 6, to - 1);
+                String end = nextInput.substring(to + 4);
+
+                if (description.isBlank() || start.isBlank() || end.isBlank()) {
+                    throw new InvalidEventException();
+                } else {
+                    arr.add(0, "event");
+                    arr.add(1, description);
+                    arr.add(2, start);
+                    arr.add(3, end);
+                }
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new InvalidEventException();
+            }
+        } else {
+            throw new InvalidCommandException();
+        }
+        return arr;
+    }
+
+    /**
+     * Parses input for markTaskStatusTrue method. If input is appropriate, returns an integer
+     * representing the number of the task to be marked.
+     *
+     * @param input The user input.
+     * @returns The number of the task.
+     * @throws InvalidTaskNumberException If there is no task with the task number in the taskList.
+     * @throws InvalidCommandException If format of command entered is invalid.
+     */
+    public int parseMarkTaskStatusTrue(String input) throws InvalidTaskNumberException, InvalidCommandException {
```

Comment:
Perhaps your method names could be polished to a shorter one? Such a long name is somewhat tedious. Same issues occur in the following code.